packages/ocl/src/Util.js

```diff
@@ -0,0 +1,106 @@
+import { composeNextState } from '@openfn/language-common';
+
+export function buildUrl(configuration, options) {
+  const {
+    ownerId,
+    ownerType,
+    repository,
+    repositoryId,
+    version,
+    expansions,
+    expansionId,
+    content,
+    query,
+    ...rest
+  } = options;
+
+  const { baseUrl } = configuration;
+
+  const urlParts = [
+    baseUrl,
+    ownerType,
+    ownerId,
+    repository,
+    repositoryId,
+    version,
+    expansions,
+    expansionId,
+    content,
+  ].filter(urlPart => urlPart);
+  const url = urlParts.join('/');
+
+  return { url, query: { ...query, ...rest } };
+}
+
+export function isObjectEmpty(objectName) {
+  return (
+    objectName &&
+    Object.keys(objectName).length === 0 &&
+    objectName.constructor === Object
+  );
+}
+
+export function handleResponse(response, state, callback) {
+  const nextState = {
+    ...composeNextState(state, response),
+    response,
+  };
+  if (callback) return callback(nextState);
+  return nextState;
+}
+
+export function handleError(error) {
+  throw error;
+}
+
+export function handleEmptyResponse(response) {
+  const { method, url, status, data } = response;
+  const responseString = [
+    `Request: ${method} "${url}"`,
+    `Status: ${status}`,
+    `Retrieved "${data.length}" data`,
+  ].join('\n∟ ');
+
+  throw new Error(`at ${new Date()} \n ${responseString}`);
+}
+
+export function handleResponseError(response, method) {
```

Comment:
this feels significantly more complex than the example we agreed on earlier. i'm OK with it if there's some important reason. what's the benefit of creating two different functions here?

Reply:
`handleEmptyResponse()` is for `status: 200`, `result: 0`, i couldn't get the `results` from `response`.  https://developer.mozilla.org/en-US/docs/Web/API/Response. The body returned in `response` object is a readable stream not a result object. 

I have modified the `handleResponseError(response,results, method)` that way i don't need `hanldeEmptyReponse()` function